include/nanobind/nb_class.h

```diff
@@ -62,10 +62,12 @@ enum class type_flags : uint32_t {
     is_arithmetic            = (1 << 16),
 
     /// Is the number type underlying the enumeration signed?
-    is_signed                = (1 << 17)
+    is_signed                = (1 << 17),
 
-    // One more flag bits available (18) without needing
-    // a larger reorganization
+    /// Does the type implement a custom __new__ operator?
+    has_new                  = (1 << 18)
+
+    // No more bits bits available without needing a larger reorganization
```

Comment:
This will stall the flag enums PR, since we're now out of bits for `is_flag`, right?

Reply:
Addressed by 72eb1b236326d77d0ff90b39b69312f2239f1527.

Reply:
Thank you!